optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -1152,6 +1157,42 @@ def to(self, device: Union[torch.device, str, int]):
         return self
 
 
+    @staticmethod
+    def infer_onnx_filename(
+        model_name_or_path: Union[str, Path],
+        pattern: str,
+        argument_name: str,
+        subfolder: str = "",
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        fail_if_not_found: bool = True,
+    ) -> str:
+        onnx_files = find_files_matching_pattern(
+            model_name_or_path,
+            pattern,
+            glob_pattern="**/*.onnx",
+            subfolder=subfolder,
+            use_auth_token=use_auth_token,
+            revision=revision,
+        )
+
+        path = model_name_or_path
+        if subfolder != "":
+            path = f"{path}/{subfolder}"
+
+        if len(onnx_files) == 0:
+            if fail_if_not_found:
+                raise FileNotFoundError(f"Could not find any ONNX model file in {path}")
+            return None
+        elif len(onnx_files) > 1:
+            if argument_name is not None:
+                raise RuntimeError(
+                    f"Too many ONNX model files were found in {path}, specify which one to load by using the "
+                    f"{argument_name} argument."
+                )
+        return onnx_files[0].parent.name + "/" + onnx_files[0].name
```

Comment:
Why not use path join instead? Also it seems that you want to produce the "parent/filename" string, I guess you can do using Path directly.

Reply:
You can create a path doing `Path(x, y, z)` if that's what you want, with `x, y, z` either strings or `Path`

Reply:
I was hesitant of changing `infer_onnx_filename`'s behaviour, but I just noticed that it's only used in `modeling_decoder.py` and `modeling_seq2seq.py` so yes I think that's a very good suggestion, thanks!